src/mixins/canvas_grouping.mixin.js

```diff
@@ -140,14 +140,19 @@
             (allowIntersect && currentObject.containsPoint(selectionX2Y2))
         ) {
           group.push(currentObject);
-
           // only add one object if it's a click
           if (isClick) {
             break;
           }
         }
       }
 
+      if (group.length > 1) {
+        group = group.filter(function(object) {
+          return !object.onSelect({ e: e });
```

Comment:
what about wrapping `group.push(currentObject)` in the if with this condition?

```
if (!object.onSelect({ e: e })) {
  group.push(currentObject);
}
```

so we avoid the other if and the filter operation, just because is 5 lines extra not because i do not like it.

Reply:
That was my initial implementation. However, if the selection only intersected one object the _onSelect()_ method would be called 2 times. one in _this method_ and another on  __setActiveObject()_. That requires changing the __setActiveObject()_ to have a flag to not call the _onSelect()_ if we are 'coming' from the __onCollectObjects()_. I was reluctant to change as it could impact many operations. However, if you like that aproach better let me know and I'll do a refactoring

Reply:
no, at this point this solution is better.

We still need tests. Can you write them?

Reply:
yes, i wasn't able this weekend, but I'll work on it this week. I'll keep you updated.

Reply:
@asturur Already implemented the tests. If you need anything more just let me know